DS/DsAssignment4/PrintApplicationTest.java

```diff
@@ -0,0 +1,65 @@
+import static org.junit.Assert.*;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class PrintApplicationTest {
+
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+	}
+
+	@Test
+	public void testProcessRequestIfPrinterIsBuzy() {
+
+		PrintApplication application = new PrintApplication();
+		application.printerStatus = false;// making printer busy
+		String expected = "you are in queue...plz wait for some time";
+		Task task = new Task(4);
```

Comment:
Why would I create a task? Shouldn't it be internal. 
I was expecting a better oop solution here. Consider you have a 4 classes: GStudent, Professor, Chairman and UGStudent. All these classes/objects have their priority already defined inside their class. Now in the test, instead of creating a new task with priority every time I would simply do the following:-

GStudent st = new GStudent();
application.processRequest(st);

Professor pf = new Professor ();
application.processRequest(pf);